src/test/java/com/rackspace/ceres/app/web/WriteControllerTest.java

```diff
@@ -223,4 +229,29 @@ public void ingestWithoutMetricGroup() {
 
     verifyNoMoreInteractions(dataWriteService);
   }
+
+  @Test
+  public void ingestTimestampOutOfBounds() throws NoSuchMethodException {
+    Metric metric = new Metric()
+            .setMetric("metricA")
+            .setTags(Collections.singletonMap("os", "linux"))
+            .setTimestamp(Instant.parse("1997-01-01T00:00:00Z"))
+            .setValue(123);
+
+    BindingResult bindingResult = new MapBindingResult(new HashMap<>(), "objectName");
+    bindingResult.addError(
+            new FieldError("objectName", "timestamp", "Out of bounds!!"));
+
+    Method method = this.getClass().getMethod("ingestTimestampOutOfBounds", (Class<?>[]) null);
+    MethodParameter parameter = new MethodParameter(method, -1);
+
+    // TODO: We shouldn't have to force the exception here. Why isn't the IngestBoundValidator doing it's job??
```

Comment:
The reason its not triggering validation is validation is occurring when Flux starts emitting data and this happens when `metrics.flatMap(tuple -> ingest(tuple.getT1(), tuple.getT2()));` is called inside DataWriteService. But the test  has mockbean for this class and hence no calls to Flux to emit data. To see the behavior we can comment this  `when` call and adjust WriteController#putMetrics method as below:

```
final Flux<Metric> results = metrics
            .map(metric -> {
              String tenant;
              filterMetricTags(metric);
              if (StringUtils.hasText(tenantHeader)) {
                tenant = tenantHeader;
              } else {
                tenant = metric.getTags().remove(appProperties.getTenantTag());
              }
              return Tuples.of(tenant, metric);
            }).flatMap( tuple -> dataWriteService.ingest(tuple.getT1(), tuple.getT2()));
```
This block of code iterates over flux which trigger validations.

This optional change if we want to avoid explicitly raising exception from mockbean implementation.